CONTRIBUTING.md

```diff
@@ -357,19 +397,64 @@ Beside general good practices for regular expresions, keep in mind:
-|               |              |                     | `[\\.\\d]+,[\\d]{2}` |
-|               | 74'120.00    | `[\\d.']+`          | `[\\.'\\d]+` |
-|               | 20 120.00    | `[\\d.\\s]+`        | `[\\.\\d\\s]+` |
-| Currency      | EUR          | `\\w+`              | `[A-Z]{3}` |
-|               |              |                     | `[\\w]{3}` |
-| Currency      | € or $       | `\\D`               | `\\p{Sc}` |
+Keep in mind that the regular expressions work against text that is automatically created from PDF files. Due to the nature of the process, there can always be slight differences in the text files. The following table collects the regular expressions that **worked well** to match typical values.
+
+| Value	| Example	| Not Helpful	| Works Well	|
+| :-----------	| :-----------	| :------------	| :-----------	|
+| Date			| 01.01.1970	| `\\d+.\\d+.\\d{4}`	| `[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}`			|
+|				| 1.1.1970		| `\\d+.\\d+.\\d{4}`	| `[\\d]{1,2}\\.[\\d]{1,2}\\.[\\d]{4}`	|
+| Time			| 12:01			| `\\d+:\\d+`			| `[\\d]{2}\\:[\\d]{2}}`						|
+| ISIN			| IE00BKM4GZ66	| `\\w+`					| `[A-Z]{2}[A-Z0-9]{9}[0-9]`					|
+| WKN			| A111X9		| `\\w+`					| `[A-Z0-9]{6}`									|
+| Amount		| 751,68		| `[\\d,.]+`				| `[\\.,\\d]+`									|
+|				|				|							| `[\\.\\d]+,[\\d]{2}`							|
+|				| 74'120.00		| `[\\d.']+`				| `[\\.'\\d]+`									|
+|				| 20 120.00		| `[\\d.\\s]+`			| `[\\.\\d\\s]+`								|
+| Currency      | EUR			| `\\w+`					| `[A-Z]{3}`										|
+|				|				|							| `[\\w]{3}`										|
+| Currency      | € or $		| `\\D`					| `\\p{Sc}`										|
+
+
+## Trade calendar
+
+The trade calendar takes into account the weekends and regional holidays when there is no exchange trading. 
+The non-trading days of the stock exchange itself are also included.
+
+### Source Location Trade calendar
+
+Trade calendar Manager: `name.abuchen.portfolio\src\name\abuchen\portfolio\util\TradeCalendarManager.java`
+
+Trade calendar Class: `name.abuchen.portfolio\src\name\abuchen\portfolio\util\TradeCalendar.java`
+
+Holiday types: `name.abuchen.portfolio\src\name\abuchen\portfolio\util\HolidayType.java`
+
+Holiday class: `name.abuchen.portfolio\src\name\abuchen\portfolio\util\Holiday.java`
+
+Holiday name: `name.abuchen.portfolio\src\name\abuchen\portfolio\util\HolidayName.java`
+
+Tests: `name.abuchen.portfolio.tests\src\name\abuchen\portfolio\util\TradeCalendarTest.java`
+
+### Anatomy of a Trade calendar
+
+The structure of the trade calendars is as follows:
+
+* Identification
+	* `code` --> Identification nyse --> New York Stock Exchange
+* Label
+	* `description` --> Display label of trade calendar, e.g., *New York Stock Exchange*
+* Days of weekend
+	* `weekend` --> Sets the default days for the weekend in the Trade Calendar, e.g., *Saturday and Sunday*
+
+
+### Test Cases Trade calendar
+
+The test cases include trade-free days of the trade calendar.
+
+The structure of the test cases is as follows:
```

Comment:
I don’t really understand this section, neither what it is trying to say nor the purpose.

Reply:
I have rewritten the complete part of the Trade Calendar again.